Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp

```diff
@@ -954,10 +955,18 @@ namespace AzToolsFramework
 
         void PrefabIntegrationManager::SaveCurrentPrefab()
         {
-            m_prefabSaveHandler.ExecuteSavePrefabDialog(
-                s_prefabFocusInterface->GetFocusedPrefabTemplateId(s_editorEntityContextId),
-                true
-            );
+            TemplateId currentTemplateId = s_prefabFocusInterface->GetFocusedPrefabTemplateId(s_editorEntityContextId);
+
+            if (currentTemplateId == Prefab::InvalidTemplateId)
+            {
+                auto* prefabEditorEntityOwnershipInterface = AZ::Interface<AzToolsFramework::PrefabEditorEntityOwnershipInterface>::Get();
+                if (prefabEditorEntityOwnershipInterface)
+                {
+                    currentTemplateId = prefabEditorEntityOwnershipInterface->GetRootPrefabTemplateId();
+                }
+            }
```

Comment:
Since executeSavePrefabDialog returns early for an invalid template id, this part is not necessary. Besides, there should always be a focused template id. If there isn't one, asserting or doing a return is better than assuming that the user wants to save the level instead

Reply:
I can probably remove this code now. I was getting some weird AR crashes that I could not reproduce locally, but I think they were due to the interface not being initialized in some autotest setups. A return if the interface isn't set should be enough.

Reply:
Done!